Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1764,6 +1960,40 @@ void UpdateConstraint(ConstraintData data)
         r.sharedMaterials = materials;
     }
 
+    internal bool EraseInstanceInfoRecord(Identifier identifier)
+    {
+        var path = identifier.name;
+
+        var ret = false;
+        if (this.m_clientInstances.TryGetValue(path, out InstanceInfoRecord rec))
+        {
+             ret = this.m_clientInstances.Remove(path);
+             
+             DestroyImmediate(rec.renderer);
+             
+            if (onDeleteInstanceInfo != null)
+                onDeleteInstanceInfo(identifier.name);
+        }
+
+        return ret;
+    }
+
+    internal bool EraseInstancedEntityRecord(Identifier identifier)
```

Comment:
Very similar to `EraseInstanceInfoRecord`.
Perhaps we can
1. use generics to create a new function
2. pass the onDelete function to it.

to encapsulate all the common code ?